### PR TITLE
feat: Support remote recording org.apache.http

### PR DESCRIPTION
**Pull request by apotterri**
Add a hook to support the remote recording endpoint of a server based on
org.apache.http.nio.

### CODE REVIEW

**Comment by apotterri** [comment]
Initial agent jar, for debugging:
[appmap-httpcore.zip](https://github.com/applandinc/appmap-java/files/7850540/appmap-httpcore.zip)

**Comment by apotterri** [comment]
Agent jar with more debugging:
[appmap-httpcore-20220112.zip](https://github.com/applandinc/appmap-java/files/7856666/appmap-httpcore-20220112.zip)

**Comment by apotterri** [comment]
Another one for today:
[appmap-httpcore-20220112-1.zip](https://github.com/applandinc/appmap-java/files/7857161/appmap-httpcore-20220112-1.zip)

**Comment by apotterri** [comment]
Capture package and protected methods, more logging:

[appmap-httpcore-20220113.zip](https://github.com/applandinc/appmap-java/files/7890478/appmap-httpcore-20220113.zip)

**Comment by apotterri** [comment]
Hook HttpRequestHandler, too:
[appmap-httpcore-20220120.zip](https://github.com/applandinc/appmap-java/files/7904947/appmap-httpcore-20220120.zip)

**Comment by apotterri** [comment]
Hook HttpProcess, have it skip interceptors:
[appmap-httpcore-20220204.zip](https://github.com/applandinc/appmap-java/files/8002470/appmap-httpcore-20220204.zip)

File: build.gradle [hunk 1]
```diff
@@ -54,12 +54,19 @@ dependencies {
...
   implementation 'info.picocli:picocli:4.6.1'
+  implementation 'org.apache.httpcomponents:httpcore-nio:4.4.15'
+
 
   testImplementation 'junit:junit:4.12'
   testImplementation 'com.github.stefanbirkner:system-rules:1.19.0'
   testImplementation "org.mockito:mockito-core:3.+"
 }
 
+compileJava {
```
Comment (apotterri) [inline build.gradle:65]: We're building with JDK 1.8 in Travis, but these settings mean you can use any JDK.

**Review by dustinbyrne — APPROVED** [review]
Looks great!

**Pull request merged**